### ISSUE TITLE
Entity subgenerator 

### ISSUE THREAD

**Issue by fmatuszewski** [jhipster/generator-jhipster#3681]
* **Overview of the request**

Bug. 

* **Motivation for or Use Case** 

When execute entity sub generator on gateway project on imported entity and adding new relationships.
Then after re generation microserviceName and searchEngine options are removed from .jhipster file
When after that entity is re generated then it generate backend logic because microserviceName was omitted form description file.

I found this bug when I wanted to generate new relationships between entities that are in two different microservices. It is impossible to do it in backend. But relationship can be added in Gateway frontend code base.

* **Browsers and Operating System** 

n/a

* **Related issues** 

n/a

* **Suggest a Fix** 

Fix  entity sub generator [BETA] Yes, add more fields and relationships 
option

**Comment by deepu105** [comment]
I guess its working as designed. You are suppose to import an entity to gateway from a  microservice only means if you want to change the entity you should change it on the microservice and then import it back again. If you change it on the gateway it doesn't update the microservice and it will be treated as a normal gateway entity. So a fix for this would be to not offer the edit options if an entity is imported from a microservice.

**Comment by deepu105** [comment]
OK on second thought I guess it would make sense to have an option to edit the entity here.

**Comment by fmatuszewski** [comment]
What I want to do is to maintain relationship between entities in two different microservices.
What I did. I created eintity with fields for foreign keys like foreignEntityId and I now override this in frontend so I can get a combo box generated. and it works.
It is no problem for me because I can modify this field by hand but on the other hand I think it should bo possible to modify it in the forntend only to meet more complicated scienarios like this.
